CLI/src/main/java/backend/FileDownloader.java

```diff
@@ -88,7 +76,7 @@
                         downloaderThreads.add(downloader);
                         tempFiles.add(file);
                     }
-                    ProgressBarThread progressBarThread = new ProgressBarThread(fileOutputStreams, partSizes, fileName, getDir(), totalSize, downloadMetrics);
+                    ProgressBarThread progressBarThread = new ProgressBarThread(fileOutputStreams, partSizes, fileName, dir, totalSize, downloadMetrics);
```

Comment:
This work will be done when we would be implementing parallel downloading feature as now the files are downloaded sequentially and the progress bar works cleanly.

Reply:
Yes, you must create a detailed issue following the structure of the issue content as written in [feature request for application template](https://github.com/SaptarshiSarkar12/Drifty/blob/master/.github/ISSUE_TEMPLATE/feature-request-application.yaml). Remember to mention some of the most important benefits of parallel downloading with a proposed solution.

Reply:
Your proposed solution is completely wrong. I was talking about parallelly downloading multiple files.
Downloading a single file in chunks is already present in the current source code. Please re-write and update your comment there.

Reply:
Can you please elaborate the process of implementation with minute details so that it can be turned into a good-first-issue? Moreover, the parallel downloading feature must also be implemented for GUI - so, please elaborate for GUI also.

Reply:
You have not updated the issue comment with the detailed proposed solution. Please update it as soon as possible.